progress/meter.go

```diff
@@ -32,24 +33,70 @@ type ProgressMeter struct {
 	dryRun            bool
 }
 
-// NewProgressMeter creates a new ProgressMeter for the number and size of
-// files given.
-func NewProgressMeter(estFiles int, estBytes int64, dryRun bool, logPath string) *ProgressMeter {
-	logger, err := newProgressLogger(logPath)
-	if err != nil {
+type env interface {
+	Get(key string) (val string, ok bool)
+}
+
+type meterOption func(*ProgressMeter)
+
+func DryRun(dryRun bool) meterOption {
+	return func(m *ProgressMeter) {
+		m.dryRun = dryRun
+	}
+}
+
+func WithLogFile(name string) meterOption {
+	printErr := func(err string) {
 		fmt.Fprintf(os.Stderr, "Error creating progress logger: %s\n", err)
 	}
 
-	return &ProgressMeter{
-		logger:         logger,
+	return func(m *ProgressMeter) {
+		if len(name) == 0 {
+			return
+		}
+
+		if !filepath.IsAbs(name) {
+			printErr("GIT_LFS_PROGRESS must be an absolute path")
+			return
+		}
+
+		cbDir := filepath.Dir(name)
+		if err := os.MkdirAll(cbDir, 0755); err != nil {
+			printErr(err.Error())
+			return
+		}
+
+		file, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			printErr(err.Error())
+			return
+		}
+
+		m.logger.writeData = true
+		m.logger.log = file
+	}
+}
+
+func WithOSEnv(os env) meterOption {
```

Comment:
They have the same interface, but git and os envs are very different. Just made sense to make it more explicit in the method name. I was envisioning a type that accepted OS and git env values.